haystack/document_stores/search_engine.py

```diff
@@ -1627,7 +1635,46 @@ def delete_index(self, index: str):
             )
         self._delete_index(index)
 
-    def _delete_index(self, index: str):
+    def _delete_index(self, index: str, headers: Optional[Dict] = None):
         if self._index_exists(index):
-            self.client.indices.delete(index=index, ignore=[400, 404])
+            self._client_indices_delete(index=index, headers=headers)
             logger.info("Index '%s' deleted.", index)
+
+    def _client_indices_exists(self, index: str, headers: Optional[Dict]) -> bool:
+        return self.client.indices.exists(index=index, headers=headers)
+
+    def _client_indices_exists_alias(self, index: str, headers: Optional[Dict]) -> bool:
+        return self.client.indices.exists_alias(name=index, headers=headers)
+
+    def _client_count(self, index: str, body: Dict, headers: Optional[Dict]) -> Dict:
+        return self.client.count(index=index, body=body, headers=headers)
+
+    def _client_msearch(self, index: str, body: List[Dict]) -> Dict:
+        return self.client.msearch(index=index, body=body)
+
+    def _client_indices_refresh(self, index: str, headers: Optional[Dict]):
+        self.client.indices.refresh(index=index, headers=headers)
+
+    def _client_delete_by_query(self, index: str, body: Dict, headers: Optional[Dict]):
+        self.client.delete_by_query(index=index, body=body, headers=headers, ignore=[404])
+
+    def _client_indices_delete(self, index: str, headers: Optional[Dict]):
+        self.client.indices.delete(index=index, headers=headers, ignore=[400, 404])
+
+    def _client_indices_get(self, index: str, headers: Optional[Dict]) -> Dict:
+        return self.client.indices.get(index=index, headers=headers)
+
+    def _client_indices_put_mapping(self, index: str, mapping: Dict, headers: Optional[Dict]):
+        self.client.indices.put_mapping(index=index, body=mapping, headers=headers)
```

Comment:
Why all these one line methods are necessary? Can't we just call those lines in their place?

Reply:
See the comment in `elasticsearch7.py`.